test/controllers/articles_controller_test.rb

```diff
@@ -3,28 +3,86 @@
 require "test_helper"
 
 # Controller tests for news articles
-class ArticleControllerTest < FunctionalTestCase
-  def test_create_article_get
+class ArticlesControllerTest < FunctionalTestCase
+  ############ test Actions that Display data (index, show, etc.)
+
+  def test_index
+    # Prove any user can see article index
+    get(:index_articles)
+    assert(:success)
+
+    # Prove privileged user get link to create an article
+    login(users(:article_writer).login)
+    get(:index_articles)
+    assert_select("a", text: :create_article_title.l)
+  end
+
+  def test_list_articles
+    get(:list_articles)
+    assert(:success)
+  end
+
+  def test_show
+    # Prove that an actual article gets shown
+    get(:show, id: articles(:premier_article).id)
+    assert_response(:success)
+    assert_template(:show)
```

Comment:
Add a test of content here, instead of, or in addition to the template test.
Probably test that <body> include the full article text.